src/main/java/com/powsybl/openloadflow/sensi/AbstractDcSensitivityAnalysis.java

```diff
@@ -131,57 +156,60 @@ private SensitivityValue createBranchSensitivityValue(LfNetwork lfNetwork, Equat
+                    column = p.getColumn();
+                } else {
+                    Equation p = equationSystem.getEquation(lfBus.getNum(), EquationType.BUS_P).orElseThrow(IllegalStateException::new);
+                    column = p.getColumn();
                 }
-                Equation p = equationSystem.getEquation(lfBus.getNum(), EquationType.BUS_P).orElseThrow(IllegalStateException::new);
-                rhs.set(p.getColumn(), factorGroup.getIndex(), busAndInjection.getValue() / PerUnit.SB);
+                rhs.set(column, factorGroup.getIndex(), busAndInjection.getValue() / PerUnit.SB);
             }
             for (String phaseTapChangerHolderId : configuration.getPhaseTapChangerHolderIds()) {
                 LfBranch lfBranch = lfNetwork.getBranchById(phaseTapChangerHolderId);
                 Equation a1 = equationSystem.getEquation(lfBranch.getNum(), EquationType.BRANCH_ALPHA1).orElseThrow(IllegalStateException::new);
                 rhs.set(a1.getColumn(), factorGroup.getIndex(), Math.toRadians(1d));
             }
         }
-        return rhs;
     }
 
-    private Map<SensitivityVariableConfiguration, SensitivityFactorGroup> indexFactorsByVariableConfig(Network network, List<SensitivityFactor> factors, LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters) {
+    protected Map<SensitivityVariableConfiguration, SensitivityFactorGroup> indexFactorsByVariableConfig(Network network, List<SensitivityFactor> factors, LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters) {
         Map<SensitivityVariableConfiguration, SensitivityFactorGroup> factorsByVarConfig = new LinkedHashMap<>(factors.size());
         Map<String, Double> participationFactorByBus = getParticipationFactorByBus(lfNetwork, loadFlowParameters); // empty if slack is not distributed
-        participationFactorByBus.remove(lfNetwork.getSlackBus().getId()); // the injection on the slack bus will not appear in the rhs
-        participationFactorByBus.replaceAll((key, value) -> -value); // the slack distribution on a bus will be the opposite of its participation
+        participationFactorByBus.replaceAll((key, value) -> -value); // the slack distribution on a bus is the opposite of its participation factor
 
         // index factors by variable config
         for (SensitivityFactor<?, ?> factor : factors) {
             if (factor instanceof BranchFlowPerInjectionIncrease) {
                 BranchFlowPerInjectionIncrease injectionFactor = (BranchFlowPerInjectionIncrease) factor;
-                Injection<?> injection = getInjection(network, injectionFactor.getVariable().getInjectionId());
+                Injection<?> injection = getInjection(network, injectionFactor.getVariable().getInjectionId(), throwExceptionIfNullInjection());
+                if (injection == null) {
+                    LOGGER.debug("Injection {} not found in the network", injectionFactor.getVariable().getInjectionId());
+                    // FIXME
```

Comment:
I have added here a `FIXME` because after the code review I think that we need a decidate analysis of contingencies and factors to avoid treatements all over the code. This analysis will be done before the base case sensitivity analysis and before each N-k analysis. This graph analysis is needed because :
- We only support sensivity analysis on main connected component but the external API exposes the whole network. The list of contingencies provided by the external API is also over the whole network and these lead to a lot of if/else in the code ;
- We will need this kind of analysis on factors and injections involved in the slack distribution when a N-k leads to a separate connected component (connected or off load).